index.js

```diff
@@ -200,45 +205,38 @@ Dat.prototype.download = function (cb) {
     self.live = archive.live
     self.db.put('!dat!key', archive.key.toString('hex'))
 
-    archive.metadata.once('download-finished', updateTotalStats)
-
     archive.content.on('download-finished', function () {
-      if (self.stats.bytesTotal === 0) updateTotalStats() // TODO: why is this getting here with 0
       self.emit('download-finished')
     })
 
-    each(archive.list({live: archive.live}), function (data, next) {
-      var startBytes = self.stats.bytesProgress
-      archive.download(data, function (err) {
+    index({
+      start: 0,
+      feed: self.archive.metadata,
+      db: self.db,
+      live: self.live
+    }, function onentry (buf, next) {
+      var entry = entryEncoding.decode(buf)
+      if (entry.type !== 'file') return next()
+      if (archive.isEntryDownloaded(entry)) return done()
+      archive.download(entry, done)
+
+      function done (err) {
         if (err) return cb(err)
-        if (startBytes === self.stats.bytesProgress) {
-          // TODO: better way to measure progress with existing files
-          self.stats.bytesProgress += data.length
-        }
-        if (data.type === 'file') {
-          self.stats.filesProgress++
-          self.emit('file-downloaded', data)
-        }
+        self.stats.filesProgress++
```

Comment:
hmmm, sucks that you have to create another instance of `hypercore-index` here.

Also, the index will on repeated invocation start on the entry that it has last seen, which means that `self.stats.filesProgress` will be off, as it always starts at `0` (right?)